### PR TITLE
Sb/two neighbor

### PR DESCRIPTION
**Pull request by simonbrandhorst**

### CODE REVIEW

**Comment by simonbrandhorst** [comment]
@HechtiDerLachs  
In the 3 conversion functions there seems to be much code which is copy pasted. 
Could you clean that up and unify it?

File: experimental/Schemes/WeilDivisor.jl [hunk 1]
```diff
@@ -327,8 +332,15 @@ function intersect(D::WeilDivisor, E::WeilDivisor;
         result = a1*a2*_self_intersection(c1)
```
Comment (simonbrandhorst) [inline experimental/Schemes/WeilDivisor.jl]:
```suggestion
        result = result + a1*a2*_self_intersection(c1)
```

File: experimental/Schemes/WeilDivisor.jl [hunk 2]
```diff
@@ -327,8 +332,15 @@ function intersect(D::WeilDivisor, E::WeilDivisor;
...
       else
         I = c1 + c2
-        @assert dim(I) <= 0 "divisors have nontrivial self intersection"
-        result = result + a1 * a2 * colength(I, covering=covering)
+        if dim(I) > 0
+          if c1 == c2
+            result = a1*a2*_self_intersection(c1)
```
Comment (simonbrandhorst) [inline experimental/Schemes/WeilDivisor.jl]:
```suggestion
            result = result + a1*a2*_self_intersection(c1)
```

File: experimental/Schemes/elliptic_surface.jl [hunk 3]
```diff
@@ -1116,11 +1214,37 @@ function elliptic_parameter(X::EllipticSurface, F::Vector{QQFieldElem})
...
+end
+
+@doc raw"""
+  elliptic_parameter(X::EllipticSurface, F::Vector{QQFieldElem}) -> LinearSystem
+
+Return the complete linear system ``|F|``.
+
+Here `F` must be given with respect to the basis of
+`algebraic_lattice(X)` and be an isotropic nef divisor. Assumes that $X$ is
+a K3 surface.
```
Comment (simonbrandhorst) [inline experimental/Schemes/elliptic_surface.jl]:
```suggestion
Return the elliptic parameter ``u`` of the divisor class `F`. 

The input `F` must be given with respect to the basis of
`algebraic_lattice(X)` and be an isotropic nef divisor. 
This method assumes that $X$ is a K3 surface.
```

**Comment by simonbrandhorst** [comment]
From my point of view this is good to go.

**Comment by simonbrandhorst** [comment]
Tests fail. Could you please investigate @HechtiDerLachs ?

**Comment by HechtiDerLachs** [comment]
We put new assertions in which were not satisfied by our test cases.

**Review by HechtiDerLachs — APPROVED** [review]

**Pull request merged**